drivers/clock_control/clock_control_nrf_auxpll.c

```diff
@@ -67,6 +79,60 @@ static int clock_control_nrf_auxpll_off(const struct device *dev, clock_control_
 	return 0;
 }
 
+static void onoff_start_auxpll(struct onoff_manager *mgr, onoff_notify_fn notify)
+{
+	struct dev_data_auxpll *dev_data =
+		CONTAINER_OF(mgr, struct dev_data_auxpll, mgr);
+
+	dev_data->notify = notify;
+	int ret = clock_control_nrf_auxpll_on(dev_data);
+
+	if (dev_data->notify) {
+		dev_data->notify(&dev_data->mgr, ret);
+	}
```

Comment:
There's no need to store `notify` in `dev_data`.
```suggestion
	int ret = clock_control_nrf_auxpll_on(dev_data);

	notify(&dev_data->mgr, ret);
```